tuf/utils/x509.go

```diff
@@ -443,6 +454,9 @@ func EncryptPrivateKey(key data.PrivateKey, role, passphrase string) ([]byte, er
 		return nil, fmt.Errorf("unable to encrypt key - invalid PEM file produced")
 	}
 	encryptedPEMBlock.Headers["role"] = role
+	if gun != "" {
```

Comment:
This is where we set the PEM headers. As per conversation with @endophage, we decided to have no gun field instead of an empty gun in case we have a root/delegation key. So, I wrote the if block so that a `gun: ""` field doesn't get added to these keys where it would be irrelevant